Sources/ProjectDescription/Settings.swift

```diff
@@ -55,66 +55,56 @@ public enum SettingValue: ExpressibleByStringInterpolation, ExpressibleByArrayLi
 
 // MARK: - Configuration
 
-public struct Configuration: Equatable, Codable {
-    public let settings: SettingsDictionary
-    public let xcconfig: Path?
-
-    public init(settings: SettingsDictionary = [:], xcconfig: Path? = nil) {
-        self.settings = settings
-        self.xcconfig = xcconfig
-    }
-
-    public static func settings(_ settings: SettingsDictionary, xcconfig: Path? = nil) -> Configuration {
-        Configuration(settings: settings, xcconfig: xcconfig)
-    }
-}
-
 /// A custom configuration allows declaring a named build configuration along with its settings.
 ///
 /// Additionally, a custom configuration specifies the configuration variant (debug or release)
 /// to help Tuist select the most appropriate default settings.
 ///
 /// - seealso: Configuration
-public struct CustomConfiguration: Equatable, Codable {
+public struct Configuration: Equatable, Codable {
     public enum Variant: String, Codable {
         case debug
         case release
     }
 
     public let name: String
```

Comment:
We can possibly leverage `ConfigurationName` here too to allow users to have a shared type for this.

e.g.

```swift
extension ConfigurationName {
  static var beta: ConfigurationName {
     .configuration("Beta")
  }
}

let configurations: [Configuration] = [
  .release(name: .debug, xcconfig: "Configs/debug.xcconfig"), 
  .release(name: .beta, xcconfig: "Configs/beta.xcconfig"),
  .release(name: .release, xcconfig: "Configs/debug.xcconfig"),
]

let project = Project(
  name: "MyMultiConfigProject",
  settings: .settings(configurations: configurations),
 // ...
)
```

The same type can then be used in the other APIs that require only the configuration by name, e.g. in `RunAction`, `TestAction` etc...